Clean up README.md, update outdated link

ebkalderon opened the pull request:
### Changed

* Fix up capitalization, grammar, and spelling mistakes in the README.md.
* Rephrase some awkward sentences.
* Replace reference to pong example with a link to the [Games Made With Amethyst](https://community.amethyst-engine.org/t/games-made-with-amethyst/134) thread on the forum.

### Fixed

* Update outdated examples link to point to 0.10.0 examples. This tripped up @chriskrycho in the Rust 1.32 episode of New Rustacean (see [this PR](https://github.com/chriskrycho/newrustacean.com/pull/122) for the fix).

jojolepro commented on README.md:
```diff
@@ -23,11 +23,13 @@ configurable as possible.
 * Powered by a correct [Entity Component System][ECS] model.
-* Rapid prototyping with [RON][ron] files for prefabs and an abstract scripting API.
+* Rapid prototyping with [RON][ron] files for prefabs and an abstract scripting
+  API.
```
Not sure why this is now on a different line?

jojolepro commented on README.md:
```diff
@@ -37,20 +39,25 @@ These principles are what make Amethyst unique and competitive in the world of g
+Amethyst is based over a very powerful parallel [ECS][ecs] called Specs. This
+allows games built with Amethyst to maximize the available processing power to
+run as smoothly and as quickly as possible, without the headaches of
+multi-threaded programming.
```
Those line changes do not make any visual differences. Are they there so it is easier to read in a text editor?

ebkalderon replied:
Oh, that must have been because of my auto-formatting for Markdown documents in my text editor. I can revert them, if you'd like!

jojolepro commented on README.md:
```diff
@@ -86,25 +96,28 @@ $ cargo run --example name_of_example
 
-Our most advanced example is currently called pong. It is a pong game, as you may have guessed it.
+Our most advanced example is currently called pong. It is a pong game, as you
+may have guessed it.
```
We probably should have removed this completely a while ago. lol

ebkalderon replied:
True. Perhaps it would be better to link to the [Games made with Amethyst](https://community.amethyst-engine.org/t/games-made-with-amethyst/134/2) forum post? Or just remove this line entirely?

jojolepro replied:
I think linking would be the best :)

ebkalderon replied:
Sounds good to me!

ebkalderon commented:
@jojolepro Thanks for the suggestions! I've reverted the automatically inserted line breaks, removed a few extra line breaks which were already present in the README earlier, and replaced the pong example reference with a link to the _Games made with Amethyst_ thread on the forum. I've updated the PR description as well to reflect this.

ebkalderon commented:
Thanks everyone! Merging.

bors r+
